graphics/font.h

```diff
@@ -112,91 +126,103 @@ class Font {
+	 * @param y       The y position where to start drawing.
+	 * @param w       Width of the text area. This can be 0 to allow for
+	 *                obtaining the whole bounding box for a string. Note that this
+	 *                does not work with an align different than kTextAlignLeft or
+	 *                with @p useEllipsis.
+	 * @param align   Text alignment. This can be used to center a string
+	 *                in the given area or to align it to the right.
+	 * @param deltax  Offset to the x starting position of the string.
+	 * @param useEllipsis  Try to fit the string in the area by inserting an
+	 *                     ellipsis. Note that the default value is false for this
+	 *                     argument, unlike for drawString.
+	 *  
 	 * @return The actual area where the string is drawn.
 	 */
 	Common::Rect getBoundingBox(const Common::String &str, int x = 0, int y = 0, const int w = 0, TextAlign align = kTextAlignLeft, int deltax = 0, bool useEllipsis = false) const;
+	/** @overload */
 	Common::Rect getBoundingBox(const Common::U32String &str, int x = 0, int _y = 0, const int w = 0, TextAlign align = kTextAlignLeft, int deltax = 0, bool useEllipsis = false) const;
 
 	/**
-	 * Draw a character at a specific point on a surface.
+	 * Draw a character at a specific point on the surface.
 	 *
 	 * Note that the point describes the top left edge point where to draw
-	 * the character. This can be different from top left edge point of the
-	 * character's bounding box! For example, TTF fonts sometimes move
-	 * characters like 't' one (or more) pixels to the left to create better
-	 * visual results. To query the actual bounding box of a character use
+	 * the character. This can be different from the top left edge point of the
+	 * character's bounding box. For example, TTF fonts sometimes move
+	 * characters like 't' by one (or more) pixels to the left to create better
+	 * visual results. To query the actual bounding box of a character, use
 	 * getBoundingBox.
 	 * @see getBoundingBox
 	 *
-	 * The Font implemenation should take care of not drawing outside of the
+	 * The Font implementation should take care of not drawing outside of the
 	 * specified surface.
 	 *
-	 * @param dst The surface to drawn on.
-	 * @param chr The character to draw.
-	 * @param x   The x coordinate where to draw the character.
-	 * @param y   The y coordinate where to draw the character.
+	 * @param dst   The surface to draw on.
+	 * @param chr   The character to draw.
+	 * @param x     The x coordinate where to draw the character.
+	 * @param y     The y coordinate where to draw the character.
 	 * @param color The color of the character.
 	 */
 	virtual void drawChar(Surface *dst, uint32 chr, int x, int y, uint32 color) const = 0;
+	/** @overload */
 	void drawChar(ManagedSurface *dst, uint32 chr, int x, int y, uint32 color) const;
 
-	// TODO: Add doxygen comments to this
+	/**
+	 * @todo Doc required!
```

Comment:
All the `drawString` functions below, as their name suggest, draw the given `str` string to the given `dst` surface.
The arguments are the same ones as in the `getBoundingBox` above, with a few additional ones:
- `dst` The surface on which to draw the string.
- `str`  The string to draw.
- `x`    The x position where to start drawing.
- `y`    The y position where to start drawing.
- `w`   The width of the text area.
- `color` The colour with which to draw the string.
- `align` The text alignment. This can be used to center the string in the given area or to align it to the right.
- `deltax` Offset to the x starting position of the string.
- `useEllipsis` Use ellipsis if needed to fit the string in the area.